roombapy/roomba.py

```diff
@@ -179,9 +183,14 @@ def periodic_connection(self):
             return
         self.periodic_connection_running = True
         while not self.stop_connection:
-            if self._connect():
-                time.sleep(self.periodic_connection_duration)
-                self.client.disconnect()
+            try:
+                if self._connect():
```

Comment:
Actually there is no need to check the return value of `self._connect` and `_connect` don't even need to return a boolean since it always raises a `RoombaConnectionError` if it fails.

Reply:
Thanks for the pull request I'll have a look if this works!